hazelcast/src/main/java/com/hazelcast/nio/serialization/PortableWriter.java

```diff
@@ -105,16 +112,17 @@
      * @param value     int value to be written
      * @throws IOException in case of any exceptional case
      */
-    void writeShort(String fieldName, short value) throws IOException;
+    void writeShort(@Nonnull String fieldName, short value) throws IOException;
 
     /**
      * Writes a Portable.
+     * Use {@link #writeNullPortable(String, int, int)} to write a {@code null} Portable
      *
      * @param fieldName name of the field
      * @param portable  Portable to be written
      * @throws IOException in case of any exceptional case
```

Comment:
nit: `ClassDefinitionWriter`'s implementation of `writePortable`/`writeNullPortable` also throws `HazelcastSerializationException`. Not sure if we want to mention this fact here.